appeals/web/src/server/appeals/appeal-details/appeal-timetables/appeal-timetables.mapper.js

```diff
@@ -28,87 +36,80 @@ export const routeToObjectMapper = {
 
 /**
  * @param {import('./appeal-timetables.service.js').AppealTimetables} appealTimetables
- * @param { 'finalCommentReviewDate' | 'issueDeterminationDate' | 'lpaQuestionnaireDueDate' | 'statementReviewDate' } timetableProperty
- * @returns {AppealTimetablesMap | undefined}
+ * @param {AppealTimetableType} timetableType
+ * @param {Appeal} appealDetails
+ * @returns {PageContent}
  */
-export const appealTimetablesMapper = (appealTimetables, timetableProperty) => {
-	const currentDueDateIso = appealTimetables && appealTimetables[timetableProperty];
+export const mapUpdateDueDatePage = (appealTimetables, timetableType, appealDetails) => {
+	const currentDueDateIso = appealTimetables && appealTimetables[timetableType];
 	const currentDueDate = currentDueDateIso && dateToDisplayDate(currentDueDateIso);
 	const changeOrScheduleText = currentDueDate ? 'Change' : 'Schedule';
+	const timetableTypeText = getTimetableTypeText(timetableType);
 
-	if (timetableProperty === 'finalCommentReviewDate') {
-		return {
-			sideNote:
-				currentDueDate && `The current due date for the final comment review is ${currentDueDate}`,
-			page: {
-				title: 'Final comment review due date',
-				text: `${changeOrScheduleText} final comment review due date`
-			},
-			confirmation: {
-				title: 'Final comment review due date updated',
-				preTitle: 'The due date for the final comment review due date has been updated.',
-				rows: [
-					{
-						text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
-					}
-				]
-			}
-		};
-	} else if (timetableProperty === 'issueDeterminationDate') {
-		return {
-			sideNote:
-				currentDueDate && `The current due date for the issue determination is ${currentDueDate}`,
-			page: {
-				title: 'Issue determination due date',
-				text: `${changeOrScheduleText} issue determination due date`
-			},
-			confirmation: {
-				title: 'Issue determination due date updated',
-				preTitle: 'The due date for the issue determination has been updated.',
-				rows: [
-					{
-						text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
-					}
-				]
+	/** @type {PageContent} */
+	const pageContent = {
+		title: `Update${timetableTypeText ? ' ' + capitalize(timetableTypeText) : ''} due date`,
+		backLinkUrl: `/appeals-service/appeal-details/${appealDetails.appealId}`,
+		preHeading: `Appeal ${appealShortReference(appealDetails.appealReference)}`,
+		heading: `${changeOrScheduleText} ${timetableTypeText} due date`,
+		submitButtonText: 'Continue',
+		pageComponents: []
+	};
+
+	if (currentDueDate) {
+		pageContent.pageComponents?.push({
+			type: 'inset-text',
+			parameters: {
+				text: `The current due date for the ${timetableTypeText} is ${currentDueDate}`,
```

Comment:
I looked into doing this a while back and came to the conclusion it wasn't worth it, however it might be good to re-assess. However I think this should be done in its own PR so this doesn't become the never-ending "refactor everything" PR